Unable to Backup & Restore (Error)

### Steps to reproduce

Edit 1. I'm using app without signing in
1. I had installed the fdroid version (0.9.0).
2. After seeing recent github update I backed up my data.
Settings > Backup & Restore > App Backup >  checked all entried > Backup
3. Then uninstalled that version (0.9.0).
4. Now, Installed 0.9.1 version from github.
5. Restored that [backed up] data.
Settings > Backup & Restore > App Backup > Restore
6. Got this error.


**Edit 1**. I'm using app without signing in
**Edit 2.** Now I installed that 0.9.0 version from fdroid from which I earlier made the backup. I can't restore my data even to this app from which I made the backup.
Is there any solution or all my data (subscriptions+playlists) is lost ?






java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	... 1 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `com.github.libretube.db.obj.LocalPlaylistWithVideos` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (StringReader); line: 1, column: 42] (through reference chain: com.github.libretube.obj.BackupFile["localPlaylists"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:5)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:5)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:74)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:90)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:773)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:6)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:10)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:27)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:31)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:3)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:171)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:86)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:15)
	at com.github.libretube.ui.preferences.BackupRestoreSettings$$ExternalSyntheticLambda7.onActivityResult(R8$$SyntheticClass:94)
	at androidx.activity.result.ActivityResultRegistry$1.onStateChanged(ActivityResultRegistry.java:66)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:18)
	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:329)
	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:51)
	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:10)
	at androidx.fragment.app.FragmentStateManager.start(FragmentStateManager.java:57)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:69)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:58)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:34)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:58)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1435)
	at android.app.Activity.performStart(Activity.java:8018)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3457)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7664)
	... 3 more
    

### Expected behavior

It should restore all my [locally stored] subscriptions and playlists

### Actual behavior

It gave me above mentioned error.

### LibreTube version

0.9.1

### Android version

Android 12

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/db/obj/LocalPlaylist.kt
+++ b/app/src/main/java/com/github/libretube/db/obj/LocalPlaylist.kt
@@ -7,6 +7,6 @@ import androidx.room.PrimaryKey
 data class LocalPlaylist(
     @PrimaryKey(autoGenerate = true)
     val id: Int = 0,
-    var name: String,
-    var thumbnailUrl: String
+    var name: String = "",
+    var thumbnailUrl: String = ""
 )

--- a/app/src/main/java/com/github/libretube/db/obj/LocalPlaylistWithVideos.kt
+++ b/app/src/main/java/com/github/libretube/db/obj/LocalPlaylistWithVideos.kt
@@ -4,10 +4,10 @@ import androidx.room.Embedded
 import androidx.room.Relation
 
 data class LocalPlaylistWithVideos(
-    @Embedded val playlist: LocalPlaylist,
+    @Embedded val playlist: LocalPlaylist = LocalPlaylist(),
     @Relation(
         parentColumn = "id",
         entityColumn = "playlistId"
     )
-    val videos: List<LocalPlaylistItem>
+    val videos: List<LocalPlaylistItem> = listOf()
 )

